src/resources/MachineLearning/MachineLearningInterfaces.ts

```diff
@@ -24,7 +24,9 @@ export interface CommerceSupport {
 }
 
 export interface IndexExport {
-    query: string;
+    query?: string;
+    sources?: string[];
+    filterConditions?: FilterConditions[];
```

Comment:
Do you know where those parameters are documented? I was just trying to validate that they are correctly typed here. Looked in Swagger but `extraConfig` doesn't seem to document any properties, including `indexExport`.

Reply:
Thats a good question, I checked the Swagger documentation and the `extraConfig` does not have any documented properties, for any model it can be just a JSON object. The idea is the `extraConfig` is where any relevant information related to model can be stored. The interface we maintain on platform-client is a list of known properties in `extraConfig`, but it is not exhaustive, hence the `[key: string]: any;` I will check with the team and see if we will documentations to `extraConfig`. But for now, I think it is ok to have it as it is.

Reply:
> The interface we maintain on platform-client is a list of known properties in extraConfig, but it is not exhaustive, hence the [key: string]: any;

So the `[key: string]: any;` fits what is in Swagger right now, so that's fine. But I was more wondering about the "known" properties as you called them...if they are know properties why aren't they documented? And since they are not documented, would it be more appropriate to maintain those interfaces in the admin directly?

Reply:
Well, there is a bit of long story behind it, but you have a good point, these interfaces can be declared in adui directly, it will be easier to maintain too. I changed the declaration of ExtraConfig to `[key: string]: any;` and removed `IndexExport`

Reply:
Alright cool. Last thing though, since the `IndexExport` interface was removed, we might want to identify the commit as breaking.

Reply:
Yep! In my last commit i marked it as breaking: 
https://github.com/coveo/platform-client/pull/798/commits/862ebe6965a4661167c5d8e53e38fa5d0c1dea20

i think with the commit message it should be identified as BREAKING, i used `npm run commit-cli`

Reply:
Thank you !